ci-operator/step-registry/azure/provision/bastionhost/azure-provision-bastionhost-commands.sh

```diff
@@ -84,6 +84,7 @@ else
   bastion_rg="${RESOURCE_GROUP}"
```

Comment:
Maybe we do not have to call `azure-provision-resourcegroup` step in this case. We can run a common IPI install, after that, we can get resource group info from metadata.json.

The priority would be like RESOURCE_GROUP > ${SHARED_DIR}/resourcegroup > metadata.json

And maybe we can exit 0 if did not find any resource group info after the above query search.

Reply:
okay, and there might have product bug when creating cluster in existing resource group.
will change to use "RESOURCE_GROUP > ${SHARED_DIR}/resourcegroup > metadata.json".